server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceReconciler.java

```diff
@@ -512,7 +505,8 @@ private void moveShards() {
             }
         }
 
-        private AllocationStats balance() {
+        private DesiredBalanceMetrics.AllocationStats balance() {
+            // Check if rebalancing is disabled.
```

Comment:
I wonder if it's worth documenting this, perhaps even in more detail on the `AllocationDeciders#canRebalance` method?